src/components/dashboards/Client-dashboard.tsx

```diff
@@ -1,245 +1,253 @@
 
-  const [requestsForUser, setRequestsForUser] =
-    useState<UserResidentRequestsApiResponse>();
-
+  const [requestsForUser, setRequestsForUser] = useState<any[]>([]);
   const [isLoading, setIsLoading] = useState(true);
-  const [modalOpen, setModalOpen] = useState(false);
-  const [requestIdToCancel, setRequestIdToCancel] = useState<null | string>(
-    null
-  );
-  const [apiError, setApiError] = useState<null | string>(null);
+  const [selectedRequest, setSelectedRequest] = useState<any | null>(null);
+  const prevRequestsRef = useRef<any[]>([]);
+  const [isFetching, setIsFetching] = useState(false);
+  const [isModalClosing, setIsModalClosing] = useState(false);
+
+  const handleCloseModal = () => {
+    setIsModalClosing(true);
+    setTimeout(() => {
+      setSelectedRequest(null);
+      setIsModalClosing(false);
+    }, 300);
+  };
+
+  useEffect(() => {
+    if (status === "unauthenticated") {
+      router.replace("/authenticate");
+    }
+  }, [status, router]);
+
   useEffect(() => {
     if (session?.user?.isAdmin) {
       router.replace("/dashboard/admin");
+      return;
     }
-  }, [session?.user?.isAdmin, router]);
 
-  useEffect(() => {
-    // Fetch the residents requests
     if (session?.user?.id) {
       fetchResidentRequestsForUser();
     }
+  }, [session?.user]);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      if (session?.user?.id && !session?.user?.isAdmin) {
+        fetchResidentRequestsForUser(true);
+      }
+    }, 3000);
+    return () => clearInterval(interval);
   }, [session?.user?.id]);
 
-  const fetchResidentRequestsForUser = async () => {
+  const fetchResidentRequestsForUser = async (isPolling = false) => {
     try {
-      setIsLoading(true);
+      if (!session || !session.user?.id || session.user.isAdmin) {
+        return;
+      }
+
+      if (isFetching) return;
+      setIsFetching(true);
+
       const res =
         await ResidentRequestService.getResidentRequestsByAuthenticatedUser(
-          session!.user!.id!
+          session.user.id
         );
 
       const sortedRequests = res.requests.sort((a, b) => {
         return (
           new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime()
         );
       });
-      setRequestsForUser({ ...res, requests: sortedRequests });
-      setIsLoading(false);
+
+      if (prevRequestsRef.current) {
+        detectStatusChanges(prevRequestsRef.current, sortedRequests);
+      }
+
+      prevRequestsRef.current = sortedRequests;
+      setRequestsForUser(sortedRequests);
     } catch (error) {
-      console.error(error);
-      setApiError("Failed to fetch requests");
+      console.error("Error fetching resident requests:", error);
+      if (!isPolling) {
+        toast.error("Failed to fetch requests. Please try again.");
+      }
+    } finally {
+      setIsFetching(false);
       setIsLoading(false);
     }
   };
 
+  const detectStatusChanges = (prevRequests: any[], currentRequests: any[]) => {
```

Comment:
Import ResidentRequest type from prisma
```suggestion
  const detectStatusChanges = (prevRequests: ResidentRequest[], currentRequests: ResidentRequest[]) => {
```